Source/WebCore/dom/KeyboardEvent.cpp

```diff
@@ -208,6 +208,11 @@ int KeyboardEvent::keyCode() const
     return charCode();
 }
 
+int KeyboardEvent::keyCodeAsKeyDown() const
+{
+    return windowsVirtualKeyCodeWithoutLocation(m_underlyingPlatformEvent->unmodifiedWindowsVirtualKeyCode());
```

Comment:
We should probably assert that `m_underlyingPlatformEvent` is in fact keydown.

Reply:
no, I meant that we should type-check `m_underlyingPlatformEvent`. `type()` is `keypress` but `m_underlyingPlatformEvent->type()` is `keydown`, right?